jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java

```diff
@@ -80,28 +81,27 @@ public void isApplicableHasGradleWarPluginShouldReturnTrue() {
     assertThat(result).isTrue();
   }
 
-  @Test(expected = IllegalArgumentException.class)
-  public void customizeDoesNotSupportS2iBuildShouldThrowException() {
+  @Test
+  void customizeDoesNotSupportS2iBuildShouldThrowException() {
     // Given
     final Properties projectProperties = new Properties();
     projectProperties.put("jkube.generator.from", "image-to-trigger-custom-app-server-handler");
     when(generatorContext.getRuntimeMode()).thenReturn(RuntimeMode.OPENSHIFT);
     when(generatorContext.getStrategy()).thenReturn(JKubeBuildStrategy.s2i);
     when(generatorContext.getProject().getProperties()).thenReturn(projectProperties);
 
-    // When
-    new WebAppGenerator(generatorContext).customize(Collections.emptyList(), false);
-    // Then - Exception thrown
-    fail();
+    // When & Then - Exception thrown
```

Comment:
Maybe use a proper assertion for the exception.
If we have a customized message, it might be a good idea to check we're actually throwing that exception and not something else.